cli/command/service/list.go

```diff
@@ -127,14 +128,23 @@ func printTable(out io.Writer, services []swarm.Service, running, tasksNoShutdow
 			mode = "global"
 			replicas = fmt.Sprintf("%d/%d", running[service.ID], tasksNoShutdown[service.ID])
 		}
+		image := service.Spec.TaskTemplate.ContainerSpec.Image
+		ref, err := distreference.ParseNamed(image)
+		if err == nil {
+			// update image string for display
+			// the type conversion always works because an image is always tagged
+			namedTagged, _ := ref.(distreference.NamedTagged)
```

Comment:
This type assertion may fail. Images do not always have a tag.

This is probably why some tests are failing.

You shouldn't ignore the second value from the type assertion:

```go
namedTagged, ok := ref.(distreference.NamedTagged)
```

and the use of `Name` and `Tag` should be guarded by `if ok`.